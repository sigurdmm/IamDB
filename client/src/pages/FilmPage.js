import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { fetchMediaById, addComment } from '../modules/media/actions';
import CoverDisplay from '../components/CoverDisplay/index';
import './FilmPage.less';
import MediaInformation from '../components/MediaInformation';
import CommentList from '../components/CommentList';

export class FilmPage extends React.Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.number.isRequired,
      }),
    }),
    fetchMediaById: PropTypes.func.isRequired,
    addComment: PropTypes.func.isRequired,
    detailedMedia: PropTypes.shape({
      id: PropTypes.any.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string,
      rating: PropTypes.number,
      // In unix timestamp
      released: PropTypes.any,
      actors: PropTypes.array,
      director: PropTypes.string,
      thumbnails: PropTypes.shape({
        small: PropTypes.string,
        large: PropTypes.string,
      }),
      type: PropTypes.oneOf(['movie', 'series']),
    }),
    loading: PropTypes.bool,
    error: PropTypes.string,
  };

  componentDidMount() {
    this.props.fetchMediaById(this.props.match.params.id);
  }

  addComment = comment => this.props.addComment(this.props.detailedMedia.id, comment);

  render() {
    const { detailedMedia, loading, error } = this.props;

    if (loading) {
      return <div>
        Laster inn
      </div>;
    }

    if (error) {
      return <div className='error__container'>
        Kunne ikke finne informasjon
      </div>;
    }

    return <div className='media__container'>
      <MediaInformation details={detailedMedia}/>
      <section className='description__container'>
        <h2>Description</h2>
        <p>{detailedMedia.description}</p>
      </section>
      <section>
        <h2>Actors</h2>
        <CoverDisplay onSort={false} hasSearched={true} media={detailedMedia.actors} url='/actor/'/>
      </section>
      <section className="filmpage__comments">
        <h2>Comment section</h2>
        <CommentList comments={detailedMedia.comments || []} onSubmit={this.addComment}/>
      </section>
    </div>;
  }
}

const mapStateToProps = state => ({
  detailedMedia: state.media.detailedMedia,
  loading: state.media.loading,
  error: state.media.error,
});

const actionsToProps = {
  fetchMediaById,
  addComment,
};

export default connect(mapStateToProps, actionsToProps)(FilmPage);
