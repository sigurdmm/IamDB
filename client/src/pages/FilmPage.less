.media__container {
  margin-right: 15%;
  margin-left: 15%;
}

.error__container {

}

.description__container {
  margin-top: 20px;
}
