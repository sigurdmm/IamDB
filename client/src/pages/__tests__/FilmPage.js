import { render } from 'enzyme';

// Note we're not using the default exported version
import { FilmPage } from '../FilmPage';

const mockedDetails = {
  id: -1,
  name: '',
  description: '',
  director: '',
  type: 'movie',
  thumbnails: {},
  actors: [],
};

describe('<FilmPage/>', () => {
  it('should render properly', () => {
    const spy = jest.fn();
    const wrapper = render(<FilmPage
      fetchMediaById={spy}
      addComment={() => null}
      detailedMedia={mockedDetails}
      loading={false}
      error={null}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
