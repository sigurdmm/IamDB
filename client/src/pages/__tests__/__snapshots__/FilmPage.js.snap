// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmPage/> should render properly 1`] = `
<div
  class="media__container"
>
  <div
    class="information__container"
  >
    <div
      class="mediaimage"
    >
      <img
        class="image__item"
        src="no-image-found.jpg"
      />
    </div>
    <div>
      <div
        class="title__container"
      >
        <h2 />
      </div>
      <h3>
        ‚≠êÔ∏è: 0
      </h3>
      <h3>
        üé•: movie
      </h3>
      <h3>
        üé¨: unknown
      </h3>
    </div>
  </div>
  <section
    class="description__container"
  >
    <h2>
      Description
    </h2>
    <p />
  </section>
  <section>
    <h2>
      Actors
    </h2>
    <div
      class="coverdisplay"
    />
  </section>
  <section
    class="filmpage__comments"
  >
    <h2>
      Comment section
    </h2>
    <div
      class="commentlist"
    >
      <ul
        class="commentlist__list"
      />
      <form
        class="commentform"
      >
        <label
          class="commentform__label"
          for="comment"
        >
          Comment
        </label>
        <textarea
          class="commentform__comment"
          id="comment"
          name="comment"
          rows="10"
        />
        <button
          type="submit"
        >
          Submit comment
        </button>
      </form>
    </div>
  </section>
</div>
`;

exports[`<FilmPage/> should render properly when error 1`] = `
<div
  class="error__container"
>
  Kunne ikke finne informasjon
</div>
`;

exports[`<FilmPage/> should render properly when loading 1`] = `
<div>
  Laster inn
</div>
`;
